### ISSUE TITLE
Irritating error message if datagrid has no searchable field

### ISSUE THREAD

**Issue by thomasduenser** [sulu/sulu#5227]
# Problem description

When trying to search the Datagrid _(via the search field when creating a custom module that shows entities)_, an irritating error appears if none of the list properties has the `searchability="yes"` attribute.

```
"code": 0,
"message": "[Syntax Error] line 0, col 208: Error: Expected Literal, got ')'",
```

# Proposed solution

A better error message would be helpful.

**Comment by plozmun** [comment]
Maybe you can disable search in your listBuilder with `                ->disableSearching()
`

**Comment by danrot** [comment]
@sulu/core-team I see two solutions to this problem:

1. We throw an error somewhere, if the list is searched but no field is searchable
2. We just return an empty set

I think I would go for number 1. Having search enabled but every search returns an empty set is probably confusing for both the developer and the user.

**Comment by nnatter** [comment]
Option one sounds better to me too 🙂

**Comment by alexander-schranz** [comment]
A similar error seems appear when a `searchFields=notExist` is used there I think we should also throw an error and should end in a 400 bad request status code.

**Comment by jordygroote** [comment]
Option 1 sounds good :D

**Comment by danrot** [comment]
Option 1 has been implemented in #5333
